robocrys/condense/structure.py

```diff
@@ -0,0 +1,344 @@
+            ``None``, in which case
+            :class:`pymatgen.analysis.local_env.CrystalNN` will be used.
+        mineral_matcher: A ``MineralMatcher`` instance. Defaults to ``None``
+            in which case the default ``MineralMatcher`` settings will be used.
+        use_symmetry: Whether to use symmetry to determine if structure
+            compoents and sites are equivalent. If ``False``, the site geometry
+            and bonding graph information will be used.
+        symprec: The tolerance used when determining the symmetry of
+            the structure. The symmetry is used to determine if multiple
+            sites are symmetrically equivalent. If ``use_symmetry=False`` this
+            option will be ignored.
+        use_iupac_formula (bool, optional): Whether to order formulas
+            by the iupac "electronegativity" series, defined in
+            Table VI of "Nomenclature of Inorganic Chemistry (IUPAC
+            Recommendations 2005)". This ordering effectively follows
+            the groups and rows of the periodic table, except the
+            Lanthanides, Actanides and hydrogen. If set to ``False``, the
+            elements will be ordered according to the electronegativity values.
+        use_common_formulas: Whether to use the database of common formulas.
+            The common formula will be used preferentially to the iupac or
+            reduced formula.
+    """
+
+    def __init__(self,
+                 use_conventional_cell: bool = True,
+                 near_neighbors: Optional[NearNeighbors] = None,
+                 mineral_matcher: Optional[MineralMatcher] = None,
+                 use_symmetry: bool = False,
+                 symprec: float = 0.01,
+                 simplify_molecules: bool = True,
+                 use_iupac_formula: bool = True,
+                 use_common_formulas: bool = True):
+        if not near_neighbors:
+            near_neighbors = CrystalNN()
+
+        if not mineral_matcher:
+            mineral_matcher = MineralMatcher()
+
+        self.use_conventional_cell = use_conventional_cell
+        self.near_neighbors = near_neighbors
+        self.mineral_matcher = mineral_matcher
+        self.use_symmetry = use_symmetry
+        self.symprec = symprec
+        self.simplify_molecules = simplify_molecules
+        self.use_common_formulas = use_common_formulas
+        self.use_iupac_formula = use_iupac_formula
+
+    def condense_structure(self, structure: Structure) -> Dict[str, Any]:
+        """Condenses the structure into an intermediate dict representation.
+
+        Args:
+            structure: A pymatgen structure object.
+
+        Returns:
+            The condensed structure information. The data is formatted as a
+            :obj:`dict` with a fixed set of keys. For example, the condensed
+            representation of MoS2 is shown below::
+
+                {'mineral': {'type': 'Molybdenite',
```

Comment:
maybe just give a link to the rst doc rather than reproducing it here? I'm afraid this will go out of date as we update the code